src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -497,6 +499,12 @@ namespace xclcpuemhal2 {
     if(header)
     {
```

Comment:
The xclLoadXclBin function is too long to be comprehensible.  Can it be refactored?